index/scorch/segment/zap/merge.go

```diff
@@ -562,14 +562,19 @@ func writePostings(postings *roaring.Bitmap, tfEncoder, locEncoder *chunkedIntCo
 		}
 	}
 
+	var bw int
 	tfOffset := uint64(w.Count())
-	_, err = tfEncoder.Write(w)
+	if bw, err = tfEncoder.Write(w); bw == 0 {
+		tfOffset = termNotEncoded
```

Comment:
Maybe Write(), or some variant of the Write() method, can return the offset as its result?

Also, I think you'll need to bump the version numbers (for zap?  scorch?)

Reply:
thanks Steve! Indeed. :) i went tunnel visioned on the read side earlier..

Zap version bump is needed to convey that older version won't be able to interpret this new values. (even though there is no layout/format changes)
Scorch version bump isn't needed is the thinking, looking at the current version value of 2.